ros/src/styx/bridge.py

```diff
@@ -184,16 +183,17 @@ def publish_dbw_status(self, data):
         self.publishers[Topics.Vehicle.DBWEnabled.Topic()].publish(Bool(data))
 
     def publish_camera(self, data):
-
-        self.image_count += 1
-
-        # sending only every IMAGE_DEBOUNCE-th image from simulator to the topic; doesn't work, when saving images
-        if Params.Classifier.SavingImages.Get() or self.image_count % Params.Classifier.ImageDebounce.Get() == 0:
+        # Probabilistically publish the image from the simulator:
+        not_dropped = random.uniform(0,1) >= Params.Server.ImageDropProbability.Get()
```

Comment:
The latency in the system increased drastically, I can pass on this branch only half of circle in the test mode, and only first traffic light in classifier mode. I guess the problem is in this part of code (not sure), please check.

Reply:
Ah yes, the latency increased because I reverted the drop probability to 0.0 (which is equivalent to setting DEBOUNCE = 0). You can tweak that to help with the latency on your machine. All I've done is to:
a) Move the "debounce" parameter into a config param and renamed it to "image_drop_prob", and
b) Change from dropping _fixed_ frames, to dropping the frames _stochastically_ (probabilistically).

So, if you want to have the same behavior as with DEBOUNCE=3, you can set "image_drop_prob" = "0.66", since that will drop 2 out of every 3 images. We are less likely to miss important information if we drop images stochastically than in a fixed order.

Reply:
Ok to merge?

Reply:
Yeah, verified